screen/loans/src/main/java/com/ivy/loans/loandetails/LoanDetailsViewModel.kt

```diff
@@ -449,6 +462,46 @@ class LoanDetailsViewModel @Inject constructor(
         }
     }
 
+    private fun handleChangeDate() {
+        dateTimePicker.pickDate(
+            initialDate = loanRecordModalData.value?.loanRecord?.dateTime?.let {
+                with(timeConverter) { it.toLocalDateTime().toUTC() }
+            } ?: timeProvider.utcNow()
+        ) { localDate ->
+
+            val localTime = loanRecordModalData.value?.loanRecord?.dateTime?.let {
+                with(timeConverter) { it.toLocalTime() }
+            } ?: timeProvider.localTimeNow()
+
+            updateDateTime(localDate.atTime(localTime))
+        }
+    }
+
+    private fun handleChangeTime() {
+        dateTimePicker.pickTime(
+            initialTime = loanRecordModalData.value?.loanRecord?.dateTime?.let {
+                with(timeConverter) { it.toLocalTime() }
+            } ?: timeProvider.localTimeNow()
+        ) { localTime ->
+            val localDate = loanRecordModalData.value?.loanRecord?.dateTime?.let {
+                with(timeConverter) { it.toLocalDate() }
+            } ?: timeProvider.localDateNow()
+
+            updateDateTime(localDate.atTime(localTime))
+        }
+    }
+
+    private fun updateDateTime(newDateTime: LocalDateTime) {
+        val newDateTimeUtc = with(timeConverter) { newDateTime.toUTC() }
+        loanRecordModalData.value?.let { currentData ->
+            loanRecordModalData.value = currentData.copy(
+                loanRecord = currentData.loanRecord?.copy(
+                    dateTime = newDateTimeUtc
```

Comment:
@ILIYANGERMANOV The problem is that this changes is working when this ` loanRecord = currentData.loanRecord?` is not null that means when we are in editing a record it works perfectly fine,,, but while creating a new record the time and date doesnt gets updated because `loanRecord = currentData.loanRecord?` this is null........... please see the attached vedio for detail 
 
https://github.com/user-attachments/assets/eb0e172a-0cf3-4af2-8e2e-f01781479eb2

Reply:
I don't have enough context, see how it's solved in `EditTransactionViewModel`. For example, you save the dateTime in a `mutableStateOf()` that's initialized to `record.dateTime ?: timeNow`